org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/GenerateToStringHandler.java

```diff
@@ -67,23 +75,62 @@ public static CheckToStringResponse checkToStringStatus(IType type, IProgressMon
 		if (type == null) {
 			return response;
 		}
-
 		try {
 			CompilationUnit astRoot = CoreASTProvider.getInstance().getAST(type.getCompilationUnit(), CoreASTProvider.WAIT_YES, monitor);
 			if (astRoot == null) {
 				return response;
 			}
-
 			ITypeBinding typeBinding = ASTNodes.getTypeBinding(astRoot, type);
 			if (typeBinding != null) {
 				response.type = type.getTypeQualifiedName();
-				response.fields = JdtDomModels.getDeclaredFields(typeBinding, false);
+				IVariableBinding[] fields = typeBinding.getDeclaredFields();
+				List<IVariableBinding> fieldsToBindings = new ArrayList<>();
+				for (IVariableBinding variableBinding : fields) {
+					if (!Modifier.isStatic(variableBinding.getModifiers())) {
+						fieldsToBindings.add(variableBinding);
+					}
+				}
+				fieldsToBindings.sort(new BindingComparator());
+				List<IVariableBinding> inheritedFieldsToBindings = new ArrayList<>();
+				ITypeBinding superTypeBinding = typeBinding;
+				while ((superTypeBinding = superTypeBinding.getSuperclass()) != null) {
+					for (IVariableBinding candidateField : superTypeBinding.getDeclaredFields()) {
+						if (!Modifier.isPrivate(candidateField.getModifiers()) && !Modifier.isStatic(candidateField.getModifiers()) && !JdtDomModels.contains(fieldsToBindings, candidateField)) {
+							inheritedFieldsToBindings.add(candidateField);
+						}
+					}
+				}
+				inheritedFieldsToBindings.sort(new BindingComparator());
+				List<IMethodBinding> methodsToBindings = new ArrayList<>();
+				for (IMethodBinding candidateMethod : typeBinding.getDeclaredMethods()) {
+					if (!Modifier.isStatic(candidateMethod.getModifiers()) && candidateMethod.getParameterTypes().length == 0 && !"void".equals(candidateMethod.getReturnType().getName()) && !"toString".equals(candidateMethod.getName()) //$NON-NLS-1$//$NON-NLS-2$
+							&& !"clone".equals(candidateMethod.getName())) { //$NON-NLS-1$
+						methodsToBindings.add(candidateMethod);
+					}
+				}
+				methodsToBindings.sort(new BindingComparator());
+				superTypeBinding = typeBinding;
+				List<IMethodBinding> inheritedMethodsToBindings = new ArrayList<>();
+				while ((superTypeBinding = superTypeBinding.getSuperclass()) != null) {
+					for (IMethodBinding candidateMethod : superTypeBinding.getDeclaredMethods()) {
+						if (!Modifier.isPrivate(candidateMethod.getModifiers()) && !Modifier.isStatic(candidateMethod.getModifiers()) && candidateMethod.getParameterTypes().length == 0
+								&& !"void".equals(candidateMethod.getReturnType().getName()) && !JdtDomModels.contains(methodsToBindings, candidateMethod) && !"clone".equals(candidateMethod.getName())) { //$NON-NLS-1$ //$NON-NLS-2$
+							inheritedMethodsToBindings.add(candidateMethod);
+						}
+					}
+				}
+				inheritedFieldsToBindings.sort(new BindingComparator());

```

Comment:
Typo here. I think it should be `inheritedMethodsToBindings.sort(...)`

Reply:
Fixed.